### ISSUE TITLE
[Feature Request] Detailed Progress Output

### ISSUE THREAD

**Issue by JustinGrote** [PowerShell/PSResourceGet#107]
It is noted in the blog that -Verbose and -Debug will be added, but progress was not specifically called out.
https://devblogs.microsoft.com/powershell/powershellget-3-0-preview-1/

Please add progress output in addition to -whatif support, this is immensely helpful for long running operations to allow users to determine the status. All package managers have some form of this that isn't a "wall of text" of verbose output

Here's a mockup of what I think it should look like. Output will vary depending on if the activity executes in parallel or not, I've included examples of both.

![Capture](https://user-images.githubusercontent.com/15258962/78184647-ae47bf80-741e-11ea-9878-65f2d7d61e27.gif)

```powershell
function ProgressDemo {
  if (-not $GLOBAL:modules) {
    $GLOBAL:modules = find-module az.*
  }
  

  $parentid = Get-Random
  $childid = Get-Random

  function Write-MainProgress ($Message,$percentComplete) {
    Write-Progress -id $parentID -Activity 'Install-PSResource' -PercentComplete $percentComplete -Status $Message
  }

  function Write-SubProgress ($Activity,$Message,$percentComplete) {
    Write-Progress -id $childid -parentid $parentid -Activity $Activity -Status $Message -PercentComplete $percentComplete
  }

  Write-MainProgress "Finding and resolving dependencies for $($modules.count) modules" 1
  sleep 2
  Write-MainProgress "Downloading" 15
  $i=0
  $percentage = get-random 10
  while ($percentage -lt 100) {
    Write-MainProgress "Downloading" (15 + (35 * ($percentage/100)))
    Write-Subprogress -Activity "Downloading $($modules.count) modules" -Message ($modules.foreach{$_.name,$_.version -join ' '} -join ', ') -PercentComplete $percentage
    sleep 0.2
    $percentage += get-random 5
    $i++
  }

  $i=0
  foreach ($moduleitem in $modules) {
    Write-MainProgress "Installing" (50 + (40 * ($i/$modules.count)))
    $percentage = get-random 10
    while ($percentage -lt 100) {
      Write-Subprogress -Activity "Installing $($modules.count) modules to {{MODULEDESTINATION}}" -Message (@($moduleItem).foreach{$_.name,$_.version -join ' '} -join ', ') -PercentComplete $percentage
      $percentage += get-random 20
      sleep 0.1
    }
    $i++
  }
  Write-MainProgress "Verifying" 90
  Write-Subprogress -Activity "Verifying $($modules.count) modules were installed correctly" -Message 'Processing' 0
  sleep 3
  Write-Progress -id $parentid -Activity 'Done' -Completed
}
```

**Comment by alerickson** [comment]
Thanks for the feedback @JustinGrote!  I'm currently working on flushing out the Install parameters, error handling and verbose and debug statements for the next release.  So this can certainly be incorporated into that.

**Comment by alerickson** [comment]
Addressed with release of 3.0.0-beta2
